engine/schema/src/main/java/org/apache/cloudstack/storage/datastore/db/SnapshotDataStoreDao.java

```diff
@@ -30,6 +30,8 @@
 
     List<SnapshotDataStoreVO> listByStoreId(long id, DataStoreRole role);
 
+    List<SnapshotDataStoreVO> listByStoreIdInReadyState(long id);
```

Comment:
Can you make this a generic function `listByStoreIdAndState`

Reply:
yup fixed. wasn't sure why state was mapped to ObjectInDataStoreStateMachine.State and not Snapshot.State ..

Reply:
Both track different things. `SnapshotState` is more for the end snapshot object whereas `ObjectInDataStore` tracks the status of the snapshot on a particular storage pool